### PR TITLE
Feature/cli commands refactoring AEA-490

### PR DESCRIPTION
**Pull request by panasevychol**
## Proposed changes

CLI commands refactoring
#997 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/logout.py [hunk 1]
```diff
@@ -28,8 +28,17 @@
...
 @click.command(name="logout", help="Logout from Registry account.")
 def logout():
-    """Logout from Registry account."""
+    """Logout from Registry account command."""
     click.echo("Logging out...")
+    do_logout()
+    click.echo("Successfully logged out.")
+
+
+def do_logout() -> None:
```
Comment (DavidMinarsch) [inline aea/cli/logout.py:37]: private?

File: aea/cli/login.py [hunk 2]
```diff
@@ -26,6 +26,14 @@
...
 
+@click.command(name="login", help="Login to Registry account.")
+@click.argument("username", type=str, required=True)
+@click.option("--password", type=str, required=True, prompt=True, hide_input=True)
+def login(username, password):
+    """Login to Registry account."""
+    do_login(username, password)
+
+
 def do_login(username: str, password: str):
```
Comment (DavidMinarsch) [inline aea/cli/login.py:37]: private?

File: aea/cli/register.py [hunk 3]
```diff
@@ -27,6 +27,18 @@
...
+@click.option("--password", type=str, required=True, prompt=True, hide_input=True)
+@click.option(
+    "--confirm_password", type=str, required=True, prompt=True, hide_input=True
+)
+def register(username, email, password, confirm_password):
+    """Register a new Registry account CLI command."""
+    do_register(username, email, password, confirm_password)
+
+
 def do_register(
```
Comment (DavidMinarsch) [inline aea/cli/register.py:42]: private?

File: aea/cli/run.py [hunk 4]
```diff
@@ -117,7 +67,74 @@ def run(
     click_context, connection_ids: List[PublicId], env_file: str, is_install_deps: bool
 ):
     """Run the agent."""
+    _run_aea(click_context, connection_ids, env_file, is_install_deps)
+
+
+def _run_aea(
```
Comment (DavidMinarsch) [inline aea/cli/run.py:73]: I think it should be split into _get_aea and _run_aea (latter same as before)

File: aea/cli/interact.py [hunk 5]
```diff
@@ -33,48 +34,14 @@
...
-    except KeyboardInterrupt as e:
-        raise e
-    except Exception as e:
-        print(e)
-    return envelope
+@click.command()
+def interact():
+    """Interact with a running AEA via the stub connection."""
+    click.echo("Starting AEA interaction channel...")
+    run_interaction_channel()
```
Comment (DavidMinarsch) [inline aea/cli/interact.py:41]: private?

File: aea/cli/interact.py [hunk 6]
```diff
@@ -94,36 +61,67 @@ def run():
...
+            envelope.to,
+            envelope.sender,
+            envelope.protocol_id,
+            envelope.message,
+        )
+    )
+    return message
+
+
+def try_construct_envelope() -> Optional[Envelope]:
```
Comment (DavidMinarsch) [inline aea/cli/interact.py:93]: private?

File: aea/cli/fetch.py [hunk 7]
```diff
@@ -46,8 +46,6 @@
 def fetch(click_context, public_id, alias, local):
     """Fetch Agent from Registry."""
     if local:
-        ctx = cast(Context, click_context.obj)
```
Comment (DavidMinarsch) [inline aea/cli/fetch.py:49]: Sure this can be removed?

File: aea/cli/create.py [hunk 8]
```diff
@@ -129,44 +132,55 @@ def _create_aea(
...
-@click.pass_context
-def create(click_context, agent_name, author, local, empty):
-    """Create an agent."""
-    try:
-        _check_is_parent_folders_are_aea_projects_recursively()
-    except Exception:
-        raise click.ClickException(
-            "The current folder is already an AEA project. Please move to the parent folder."
-        )
+def _crete_agent_config(ctx: Context, agent_name: str, set_author: str) -> AgentConfig:
```
Comment (DavidMinarsch) [inline aea/cli/create.py:135]: typo

Comment (DavidMinarsch) [reply aea/cli/create.py:135]: Feels like this could be a helper method in config and imported here;

File: aea/cli/login.py (hunk 2 above)
Comment (panasevychol) [reply aea/cli/login.py:37]: No. Used outside.

File: aea/cli/register.py (hunk 3 above)
Comment (panasevychol) [reply aea/cli/register.py:42]: No. Used outside.

File: aea/cli/logout.py (hunk 1 above)
Comment (panasevychol) [reply aea/cli/logout.py:37]: Just by the analogy with other `do_` methods made it public.

File: aea/cli/create.py (hunk 8 above)
Comment (panasevychol) [reply aea/cli/create.py:135]: I'm not sure it will be used anywhere else.

File: aea/cli/fetch.py (hunk 7 above)
Comment (panasevychol) [reply aea/cli/fetch.py:49]: I've moved this to _fetch_agent_locally, the only place it's used.

**Pull request merged**